Controller/ContentManagement/ViewController.php

```diff
@@ -142,16 +142,35 @@ public function editAction($id, $_format, Request $request)
                     'data-ajax-save-url' => $this->generateUrl('view.edit', ['id' => $id, '_format' => 'json']),
                 ],
                 'icon' => 'fa fa-save'
-            ])->getForm();
+            ])
+            ->add('saveAndClose', SubmitEmsType::class, [
+                'attr' => [
+                    'class' => 'btn-primary btn-sm',
+                ],
+                'icon' => 'fa fa-save'
+            ])
+            ->getForm();
 
         $form->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
             $em->persist($view);
             $em->flush();
+
+            $this->addFlash('notice', sprintf('View %s has been updated', $view->getName()));
+
+            if ($_format === 'json') {
+                return $this->render('@EMSCore/ajax/notification.json.twig', [
+                    'success' => true,
+                ]);
+            }
+
+            return $this->redirectToRoute('view.index', [
+                'type' => $view->getContentType()->getName()
+            ]);
         }
 
-        return $this->render('@EMSCore/view/edit.' . $_format . '.twig', [
+        return $this->render('@EMSCore/view/edit.html.twig', [
```

Comment:
Why would you generate an HTML page when your url contains .json at the end? This is weird behaviour.
On a .json page, you expect json, not html.

Reply:
I added a block above in order to treat the json case 6dbb28ee10671cd9cb5bcaa855da40f37286dafc